bazel_tools/nixpkgs-hermetic-cc-toolchain.patch

```diff
@@ -0,0 +1,1221 @@
++      </td>
++    </tr>
++  </tbody>
++</table>
++
+ ### nixpkgs_cc_configure
+ 
+ Tells Bazel to use compilers and linkers from Nixpkgs for the CC
+@@ -328,6 +430,15 @@ toolchain. By default, Bazel autodetects a toolchain on the current
+ `PATH`. Overriding this autodetection makes builds more hermetic and
+ is considered a best practice.
+ 
++Deprecated:
++
++Use `nixpkgs_cc_configure_hermetic` instead.
++
++This uses Bazel's autoconfigure toolchain under the hood, which is
++inhermetic. In particular, system include directories specified in the
++environment can leak in and affect the cache keys of target depending on
++the cc toolchain leading to cache misses.
++
+ Example:
+ 
+ ```bzl
+diff --git a/WORKSPACE b/WORKSPACE
+index f37ab6c..f8d40ad 100644
+--- a/WORKSPACE
++++ b/WORKSPACE
+@@ -6,7 +6,7 @@ rules_nixpkgs_dependencies()
+ 
+ load(
+     "//nixpkgs:nixpkgs.bzl",
+-    "nixpkgs_cc_configure",
++    "nixpkgs_cc_configure_hermetic",
+     "nixpkgs_git_repository",
+     "nixpkgs_local_repository",
+     "nixpkgs_package",
+@@ -134,7 +134,7 @@ nixpkgs_package(
+     repository = "@nixpkgs",
+ )
+ 
+-nixpkgs_cc_configure(repository = "@remote_nixpkgs")
++nixpkgs_cc_configure_hermetic(repository = "@remote_nixpkgs")
+ 
+ nixpkgs_python_configure(
+     python2_attribute_path = "python2",
+@@ -147,6 +147,27 @@ nixpkgs_package(
+     repository = "@remote_nixpkgs",
+ )
+ 
++nixpkgs_package(
++    name = "nixpkgs_location_expansion_test",
++    build_file_content = "exports_files(glob(['out/**']))",
++    expand_location = True,
++    nix_file = "//tests:location_expansion.nix",
++    nix_file_deps = [
++        "//:nixpkgs.json",
++        "//:nixpkgs.nix",
++        "@io_tweag_rules_nixpkgs//tests:relative_imports/nixpkgs.nix",
++    ],
++    nixopts = [
++        "--arg",
++        "attrs",
++        "{ nixpkgs_json = $(location //:nixpkgs.json); nixpkgs_nix = $(location //:nixpkgs.nix); }",
++        "--arg",
++        "relative_imports",
++        "$(location @io_tweag_rules_nixpkgs//tests:relative_imports/nixpkgs.nix)",
++    ],
++    repository = "@remote_nixpkgs",
++)
++
+ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+ 
+ http_archive(
+@@ -179,12 +200,11 @@ http_archive(
+ 
+ load(
+     "//nixpkgs:toolchains/go.bzl",
+-    "nixpkgs_go_configure"
++    "nixpkgs_go_configure",
+ )
+ 
+ nixpkgs_go_configure(repository = "@nixpkgs")
+ 
+-load("@io_bazel_rules_go//go:deps.bzl", "go_rules_dependencies", "go_register_toolchains")
++load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
+ 
+ go_rules_dependencies()
+-
+diff --git a/nixpkgs/constraints/BUILD.bazel b/nixpkgs/constraints/BUILD.bazel
+index 6662530..94678bd 100644
+--- a/nixpkgs/constraints/BUILD.bazel
++++ b/nixpkgs/constraints/BUILD.bazel
+@@ -3,7 +3,7 @@ package(default_visibility = ["//visibility:public"])
+ constraint_setting(name = "nix")
+ 
+ constraint_value(
+-    name =  "support_nix",
```

Comment:
huh, does buildifier not catch this?

Reply:
`rules_nixpkgs` CI doesn't include a buildifier step right now.